docs/doxygen/bda.md

```diff
@@ -0,0 +1,147 @@
+This approach supports all requirements: A step can perform a BDA process()
+call sending BDA data to its next step and implement the BDA process()
+method for receiving BDA data from its previous step.
+
+With this approach, the DPPP input format can remain the same: The steps
+will receive and send data in the requested format.
+
+### DPStep structure
+
+DPSteps can support BDA data using a new process() function for BDA data.
+The default implementatation for this function will throw an exception.
+The existing process() function for regular data, which is now abstract,
+should also get a default implementation that throws an exception. In code:
+
+    class DPStep {
+    public:
+      virtual bool process(DPBuffer) { throw exception; } // Process regular data.
+      virtual bool process(BDABuffer) { throw exception; } // Process BDA data.
+    };
+
+With these two process functions, a DPStep can support
+regular and/or BDA data: Existing DPSteps, which only support regular
+data will remain working. Adding support for BDA data can be done
+incrementally, one DPStep at a time.
+
+A disadvantage of this approach is that DPSteps will become larger and
+more complex: They may get two modes of operation, which
+violates the principle of having a single purpose for each class.
+However, if a DPStep becomes too complex, this design allows
+splitting it into a regular variant and a BDA variant for reducing complexity
+
+### BDA data structure
+
+A crucial point in this design is the BDABuffer which holds BDA data.
+The smallest data element in this structure is a measurement row, which
+contains the measurements for all polarisations in all channels for a single
+baseline and a single time step. Since BDA averages data for multiple channels,
+different rows may have different amounts of channels. In code:
+
+    struct Row {
+      double time;
+      double exposure;
+      Matrix<complex> data(nChannels, nPolarisations);
+      Matrix<bool> flags(nChannels, nPolarisations);
+      Matrix<float> weights(nChannels, nPolarisations);
+      Matrix<bool> fullResFlags(nChannels, nPolarisations);
+    };
+
+For reducing overhead, a BDABuffer should not contain a single Row. Using
+a variable-sized vector of rows provides a flexible structure: The BDABuffer
+size can be adjusted to the number of channels in the Rows.
+Combining and splitting buffers is also possible. In code:
+
+    class BDABuffer
+    {
+      vector<Row> itsRows; // Buffer contains a variable number of rows.
+    };
+
+This approach still has a major drawback: Each Matrix object in each row
+performs its own memory allocation, which may yield high overhead.
+A BDABuffer can mitigate this overhead using a shared memory pool for all Rows.
+
+### Steps that use data from multiple iterations.
+
+Steps that use data from multiple iterations have an internal storage
+that holds the data from previous iterations. For regular data, this internal
+storage holds a fixed number of time steps.
```

Comment:
correct -> someone might think this changes, but this is the way it is already implemented, so it doesn't change.